src/cargo/core/shell.rs

```diff
@@ -148,26 +148,55 @@ impl MultiShell {
 
 impl Shell {
     pub fn create(out: Box<Write + Send>, config: ShellConfig) -> Shell {
+        Shell {
+            terminal: Shell::get_term(out),
+            config: config,
+        }
+    }
+
+    #[cfg(any(windows))]
+    fn get_term(out: Box<Write + Send>) -> AdequateTerminal {
+        // Check if the creation of a console will succeed
+        if ::term::WinConsole::new(vec![0u8; 0]).is_ok() {
+            match ::term::WinConsole::new(out) {
+                Ok(t) => if !t.supports_color() {
+                    NoColor(Box::new(t))
+                } else {
+                    Colored(Box::new(t))
+                },
+                // This should never happens, since we checked that the creation
+                // would succeed above.
+                Err(e) => unreachable!("Could not create WinConsole: {:?}", e),
```

Comment:
Could this punt the error upwards as even though the first one succeeded the second one may still have a small window where it could fail?